Release PR for v 0.4.0

nsheff opened the pull request:
Accumulates all changes from JOSS review, plus the lifespan concept.

vreuter commented on DESCRIPTION:
```diff
@@ -1,5 +1,5 @@
 Package: simpleCache
-Version: 0.3.1
+Version: 0.3.2
```
Should the version be bumped again?

nsheff replied:
yep!

vreuter commented on NEWS:
```diff
@@ -1,17 +1,30 @@
 # Change log
 All notable changes to this project will be documented in this file.
 
+## [0.3.2] -- Unreleased
```
If this is for 0.4.0, is it the case that this is now released, or is 0.32 still the unreleased one? If this is in fact 0.4.0, should we do a changelog entry for that?

vreuter replied:
Also, should there be an entry for the "lifespan" concept, or will that be added to the changelog after doing the next release?

nsheff replied:
I guess 0.3.2 was a dev version that was never released (I'm just calling it 0.4.0 for the actual release). see actual releases here: 

https://github.com/databio/simpleCache/releases

there is an entry for lifespan in there now. I had actually changed these things but hadn't savedthat commit, nice catch.

vreuter commented on R/simpleCache.R:
```diff
@@ -38,45 +38,51 @@ NULL
-#' @param cacheName	Unique name for the cache. Be careful.
-#' @param instruction  Quoted R code to be evaluated. The returned value of this
+#' @param cacheName	A character vector for a unique name for the cache. Be careful.
+#' @param instruction  R expression (bracketed) to be evaluated. The returned value of this
```
I think this should be "braced" or "in braces."

nsheff replied:
d'oh!

vreuter commented on R/simpleCache.R:
```diff
@@ -38,45 +38,51 @@ NULL
+#'     cache creation loop).
+#' @param cacheDir Character vector specifying the directory where caches are
+#'			saved (and loaded from). Defaults to the variable set by
+#'			\code{\link[=setCacheDir]{setCacheDir()}}.
```
What's the `=` do in the bracketed expression here?

nsheff replied:
Dunno -- @vpnagraj ?

vpnagraj replied:
@vreuter @nsheff that just helps construct the link to the setCacheDir function ... without it the link would point towards a package named setCacheDir (doesn't exist) rather than the function

vreuter replied:
Ah OK. So it looks like this syntax mixes a couple of the ones in this [docs guide](http://r-pkgs.had.co.nz/man.html#text-formatting). Is this then doing something different from the options described there?

vpnagraj replied:
yeah it's a mix of those ... this vignette has an example of the exact same linking: 

https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html#links

ironically in digging up that vignette to show you the example i realized i could have been using markdown the whole time :)

nsheff commented on R/cacheDirectories.R:
```diff
@@ -19,7 +19,7 @@ setCacheDir = function(cacheDir) {
 #'
 #' Sets global variable specifying the default cache directory for
-#' simpleCacheShared() calls; this function is simply a helper alias for caching
+#' \code{\link{simpleCacheShared}} calls; this function is simply a helper alias for caching
```
should this have an `=` then since it's not a package but a function?

nsheff replied:
(@vpnagraj)

vpnagraj replied:
no i dont think so because it's linking directly to the function name ... the other example was linking to setCacheDir() not setCacheDir

nsheff commented:
Ok. @vpnagraj is this approved for a final release? Does this need to be roxygenized first? Are we good to go? (just looking for your final approval that we've addressed all the review concerns)

vpnagraj commented:
@nsheff yeah i think we've covered all the bases for the review

i'm assuming everything passes `devtools::check()` ... how about `devtools::build_win()`?

vreuter commented:
Ha I'd never seen `devtools::build_win`. I was thinking "win" like "victory." Even though it's not the same, what a great function name :laughing:

nsheff commented:
In my test (`devtools::check()`), this is still failing; 

```
Status: 1 ERROR, 1 WARNING, 2 NOTEs
See
  ‘/sfs/lustre/scratch/ns5bc/tmp/Rtmpg7Roxf/simpleCache.Rcheck/00check.log’
for details.

R CMD check results
1 error  | 0 warnings | 2 notes
checking tests ... ERROR
  Running ‘testthat.R’
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
  t1 > t0 isn't true.
  
  
  3. Failure: Cache is replaced if recreate=TRUE even if cache is fresh (@test_cache_lifespan.R#87) 
  t1 > t0 isn't true.
  
  
  testthat results ================================================================
  OK: 43 SKIPPED: 0 FAILED: 3
  1. Failure: Cache file is replaced if no lifespan is specified and recreate=TRUE (@test_cache_lifespan.R#48) 
  2. Failure: Cache is replaced if older than explicit lifespan (@test_cache_lifespan.R#74) 
  3. Failure: Cache is replaced if recreate=TRUE even if cache is fresh (@test_cache_lifespan.R#87) 

```

vpnagraj commented:
@nsheff i'm running into the same error with `devtools::check()` on my end 
@vreuter it looks like this is coming from the lifespan tests

vreuter commented:
Hmm when I pull `dev` and run `devtools::check()`, the tests succeed for me.
```
vpr9v @ cphg-5L9SYF2 ~/code/simpleCache (dev)
└─ $ ▶ R

R version 3.4.3 (2017-11-30) -- "Kite-Eating Tree"
Copyright (C) 2017 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library(devtools)
> devtools::check()
Updating simpleCache documentation
Loading simpleCache
Writing simpleCache-package.Rd
Writing simpleCache.Rd
Skipping invalid path:  .tooOld.Rd 
Deleting dot-tooOld.Rd
Setting env vars --------------------------------------------------------------
CFLAGS  : -Wall -pedantic
CXXFLAGS: -Wall -pedantic
Building simpleCache ----------------------------------------------------------
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet  \
  CMD build '/home/vpr9v/code/simpleCache' --no-resave-data --no-manual 

* checking for file ‘/home/vpr9v/code/simpleCache/DESCRIPTION’ ... OK
* preparing ‘simpleCache’:
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* building ‘simpleCache_0.4.0.tar.gz’

Setting env vars --------------------------------------------------------------
_R_CHECK_CRAN_INCOMING_USE_ASPELL_: TRUE
_R_CHECK_CRAN_INCOMING_           : FALSE
_R_CHECK_FORCE_SUGGESTS_          : FALSE
Checking simpleCache ----------------------------------------------------------
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet  \
  CMD check '/tmp/RtmpVGX2Wu/simpleCache_0.4.0.tar.gz' --as-cran --timings  \
  --no-manual 

* using log directory ‘/tmp/RtmpVGX2Wu/simpleCache.Rcheck’
* using R version 3.4.3 (2017-11-30)
* using platform: x86_64-pc-linux-gnu (64-bit)
* using session charset: UTF-8
* using options ‘--no-manual --as-cran’
* checking for file ‘simpleCache/DESCRIPTION’ ... OK
* this is package ‘simpleCache’ version ‘0.4.0’
* package encoding: UTF-8
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... NOTE
Found the following hidden files and directories:
  .idea
These were most likely included in error. See section ‘Package
structure’ in the ‘Writing R Extensions’ manual.
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘simpleCache’ can be installed ... OK
* checking installed package size ... OK
* checking package directory ... OK
* checking ‘build’ directory ... OK
* checking DESCRIPTION meta-information ... OK
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... OK
* checking Rd files ... OK
* checking Rd metadata ... OK
* checking Rd line widths ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... OK
* checking for code/documentation mismatches ... OK
* checking Rd \usage sections ... OK
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking installed files from ‘inst/doc’ ... OK
* checking files in ‘vignettes’ ... OK
* checking examples ... OK
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 OK
* checking for unstated dependencies in vignettes ... OK
* checking package vignettes in ‘inst/doc’ ... OK
* checking re-building of vignette outputs ... OK
* DONE

Status: 1 NOTE
See
  ‘/tmp/RtmpVGX2Wu/simpleCache.Rcheck/00check.log’
for details.


R CMD check results
0 errors | 0 warnings | 1 note 
checking for hidden files and directories ... NOTE
Found the following hidden files and directories:
  .idea
These were most likely included in error. See section ‘Package
structure’ in the ‘Writing R Extensions’ manual.
Warning message:
Version of roxygen2 last used with this package is 6.0.1.9000.  You only have version 6.0.1 

```

vreuter commented:
Hmm success locally, but failure on Rivanna.

nsheff commented:
Ok they pass for me now.
